fog/view/server/src/block_tracker.rs

```diff
@@ -148,20 +148,21 @@ where
             // Go over all known ingress keys and check if
             // any of them need to provide this block and have not provided it
             for rec in ingress_keys {
-                // If this ingress key isn't responsible to provide this block index, we can
-                // move on
-                if !rec.covers_block_index(next_block_index) {
+                let epoch = self.sharding_strategy.get_block_range();
+                let is_key_responsible = rec.get_block_range().overlaps(&epoch)
```

Comment:
:question: the previous attempt intentionally excluded the "end" in the check, this will now check and skip over blocks after `end`.  Just want to be sure that is desired.

Reply:
Yes, this is desired. My logic earlier was flawed.